services/provider/server/server.go

```diff
@@ -422,6 +427,21 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 	return extR, nil
 }
 
+func (s *OCSProviderServer) getCephClientAnnotation(ctx context.Context, name string) (string, error) {
+	cephClient := &rookCephv1.CephClient{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: name, Namespace: s.namespace}, cephClient)
+	if err != nil {
+		return "", fmt.Errorf("failed to get rook ceph client %s secret. %v", name, err)
+	}
+	if len(cephClient.Annotations) == 0 {
+		return "", fmt.Errorf("rook ceph client %s annotation is empty", name)
+	}
+	if cephClient.Annotations["cephUserType"] == "" {
+		return "", fmt.Errorf("rook ceph client %s annotation `cephUserType` is empty", name)
+	}
```

Comment:
Why do we need this validation here? I think if the annotation does not exist just return an empty string. 
I don't think we should break reconciliation just because of a missing annotation. 
Maybe in the future we will have other kind of users who does not have the annotation on them